libminifi/include/core/FlowConfiguration.h

```diff
@@ -162,6 +165,8 @@ class FlowConfiguration : public CoreComponent {
     }
   }
 
+  gsl::not_null<utils::ChecksumCalculator*> getChecksumCalculator() { return gsl::make_not_null(&checksum_calculator_); }
```

Comment:
Why not return a reference? `gsl::not_null<utils::ChecksumCalculator*>` is only useful in case of non-static data members in my experience, to support move semantics and similar mutations.

Reply:
Yes, good idea, that makes things simpler.